src/nvim/ui.c

```diff
@@ -327,9 +327,9 @@ void ui_line(ScreenGrid *grid, int row, int startcol, int endcol, int clearcol,
           (const sattr_T *)grid->attrs + off);
 
   if (p_wd) {  // 'writedelay': flush & delay each time.
-    int old_row = row, old_col = col;
+    int old_row = row, old_col = cursor_col;
```

Comment:
@bfredl Should this be `old_row = cursor_row` ?

Reply:
It should be `cursor_row`. It might have worked by accident, as the cursor position will most likely be changed afterwards anyway.